File: YrradiaSDL2/Common/CanvasUtilities.cs
namespace YrradiaSDL2.Common
{
    public class CanvasUtilities
    {
        public Size GetCanvasSize()
        {
            int width;
            int height;
            SDL.GetWindowSize(_.sdlDevice.m_window, out width, out height);
            var scaledWidth = (int)(width / _.gameProperties.m_resolutionScaleDown);
            var scaledHeight = (int)(height / _.gameProperties.m_resolutionScaleDown);
            return new Size(scaledWidth, scaledHeight);
        }

        public Size GetDesktopResolution()
        {
            DisplayMode displayMode;
            SDL.GetDesktopDisplayMode(0, out displayMode);
            return new(displayMode.Width, displayMode.Height);
        }

        public float GetAspectRatio()
        {
            var canvasSize = _.canvasUtilities.GetCanvasSize();
            return (float)canvasSize.Width / canvasSize.Height;
        }

        public float ConvertWidthToHeight(float width)
        {
            return width * GetAspectRatio();
        }

        public float ConvertHeightToWidth(float height)
        {
            return height / GetAspectRatio();
        }
    }
}

File: src/Common/Utilities/CanvasUtilities.cpp
#include "CanvasUtilities.hpp"
#include "Core/SDLDevice/SDLDevice.hpp"

namespace Forradia
{
    Size GetCanvasSize()
    {
        Size canvasSize;

        SDL_GetWindowSize(
            _<SDLDevice>().GetWindow().get(),
            &canvasSize.width,
            &canvasSize.height);

        return canvasSize;
    }

    float CalculateAspectRatio()
    {
        auto canvasSize{GetCanvasSize()};

        auto aspectRatio{
            static_cast<float>(canvasSize.width) / canvasSize.height};

        return aspectRatio;
    }

    float ConvertWidthToHeight(float width)
    {
        return width * CalculateAspectRatio();
    }

    float ConvertHeightToWidth(float height)
    {
        return height / CalculateAspectRatio();
    }
}